pkg/controller/factory/secret_creation.go

```diff
@@ -46,6 +55,6 @@ func CreateOrUpdateSecretWithAPIReader(c client.Client, r client.Reader, secretN
 	return nil
 }
 
-func CreateOrUpdateSecret(c client.Client, secretName string, targetNamespace string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
-	return CreateOrUpdateSecretWithAPIReader(c, c, secretName, targetNamespace, data, secretType, log)
+func CreateOrUpdateSecret(c client.Client, r client.Reader, scheme *runtime.Scheme, owner metav1.Object, targetNS string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
+	return CreateOrUpdateSecretWithAPIReader(c, r, scheme, owner, targetNS, data, secretType, log)
```

Comment:
This function should use `c` only and delegate to `CreateOrUpdateSecretWithAPIReader`  by calling it with the client for both, reader and client. If this feature is not needed anymore, feel free to delete this function
```suggestion
func CreateOrUpdateSecret(c client.Client, scheme *runtime.Scheme, owner metav1.Object, targetNS string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
	return CreateOrUpdateSecretWithAPIReader(c, c, scheme, owner, targetNS, data, secretType, log)
```